### PR TITLE
Extending CaseExpression, covering #1458

### PR DESCRIPTION
**Pull request by mgoeminne**

### CODE REVIEW

**Comment by mgoeminne** [comment]
Related issue: #1458

**Comment by wumpz** [comment]
Tests will only go into JSqlParser if they all succeed.

**Comment by mgoeminne** [comment]
The last commit is an attempt to fix #1458 , and now all tested proposed in this pull request actually pass.

File: src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt [hunk 1]
```diff
@@ -4368,7 +4368,7 @@ Expression CaseWhenExpression() #CaseWhenExpression:
 }
 {
     <K_CASE> { caseCounter++; }
-    [ switchExp=Condition() ]
+    [ switchExp=Expression() ]
```
Comment (wumpz) [inline src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt:4371]: How much does the performance degrade? Where are the tests?

Comment (mgoeminne) [reply src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt:4371]: There is no performance test for the moment. Do you have a standardized way to perform that king of tests? Do you have existing tests showing the current performances? Also, how to evaluate the tradeoff between a supposed drop in performance and a better support of the SQL language?

**Pull request merged**